bundler/lib/bundler/lockfile_parser.rb

```diff
@@ -63,18 +87,26 @@ def initialize(lockfile)
       @dependencies = {}
       @state        = nil
       @specs        = {}
+      @lockfile_path = begin
+        Bundler.default_lockfile.relative_path_from(SharedHelpers.pwd)
+      rescue GemfileNotFound
+        "Gemfile.lock"
```

Comment:
Is this new rescue fixing some issue that would be worth extracting to a separate PR? If not, is it necessary?

Reply:
necessary because I wanted to pull out the call to `Bundler.default_lockfile`, which was already being used in the conflict message, but I expanded it to include in the `source` messages for parsed checksums, but it raises when no gemfile is found (which was actually hit in a couple of tests when parsing a lockfile)

Reply:
Under which situations do we parse a lockfile without an accompanying Gemfile? I guess that happened on unit specs for the parser, but not in real life? To clarify, I'm good with this, just curious.

Reply:
yeah, that. and also my random scripts lying around that parse lockfiles :)